### ISSUE TITLE
Make progress indicators accessible

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#14465]

**Comment by jonahwilliams** [comment]
We could map these to live regions/updates frequently - this is the example use case for both flags.

**Comment by dnfield** [comment]
Looking at the platform implementation for these, they are not accessible by default (and in fact, it's kind of a pain to make them accessible - see for example https://stackoverflow.com/questions/16021624/how-can-i-get-talkback-to-read-the-android-progressbar-progress.

I think we should wrap these in semantics that are disabled by default and just have a few convenience properties on the main widget(s).  It would boil down to something like this:

```dart
Semantics(
  enabled: semanticsEnabled, // defaults to false
  label: semanticsLabel,
  liveRegion: true,
  value: semanticsValue,
  child: theProgressWidget,
);
```

**Comment by jonahwilliams** [comment]
It might be worth asking some internal teams what they do for progress indicators, to see if we can give a better default
